### PR TITLE
docs: add construct architecture diagrams

### LINKED ISSUE #102: Architectural diagram
Customers would like to see an architectural diagram of what a default RFDK deployment looks like on AWS. Using standard AWS iconography would be helpful.

---

This is a 📕 documentation issue

### PR DESCRIPTION
**Pull request by jusiskin**
Fixes #102 

## Summary

Adds architectural diagrams for the primary constructs:

**Core:**

*   `HealthMonitor`
*   `ImportedAcmCertificate`
*   `X509CertificatePem`
*   `X509CertificatePkcs12`

**Deadline:**
   * `ConfigureSpotEventPlugin`
* `Repository`
* `RenderQueue`
* `SpotEventPluginFleet`
* `UsageBasedLicensing`
* `WorkerInstanceFleet`

These are added as draw.io embedded `.svg` files which can be edited using https://app.diagrams.net/. Diagrams are stored the following path convention `packages/aws-rfdk/docs/diagrams/<package>/<ConstructName>.svg` directory of the repository.

The architecture diagrams are embedded in each module's `README.md` and each corresponding constructs' TSDoc header. They are browsable via the README files in GitHub.

The diagrams are also propagated to our API reference documentation in the construct docs pages (both the main docs page and the python-specific one) and the package overviews.

This PR adds documentation that states the expectation for keeping the diagrams updated and the process for doing so. This is linked from the `CONTRIBUTING.md` file.

## Testing

* Built the API reference documentation using a JSII build from this branch
* Confirmed the diagrams appear in the rendered HTML docs for both the main docs and python docs correctly in the following places:
   *    the package overviews (README files)
   *    the construct pages

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by horsmand — COMMENTED** [review]
This looks ready for approval for me but we are reviewing the actual diagrams externally to GitHub since the diff here doesn't actually show them. I'll hold off on the approval here until the review of the diagrams is also complete.

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import { promises as fsPromises } from 'fs';
+import * as path from 'path';
+
+/**
+ * This is the regular expression that validates that the dar.io diagram is embedded within the SVG file.
```
Comment (horsmand) [inline packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]:
```suggestion
 * This is the regular expression that validates that the draw.io diagram is embedded within the SVG file.
```

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        // GIVEN
+        const diagramContents = await fsPromises.readFile(diagram, { encoding: 'utf8' });
+
+        // THEN
+        expect(diagramContents).toMatch(DRAW_IO_EMBEDDED_XML_ATTR_REGEX);
+      });
+    });
+
+  });
+});
```
Comment (horsmand) [inline packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: Missing blank line.

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  const diagrams: string[] = [];
+
+  const listings = fs.readdirSync(dirPath);
+
+  for (const listing of listings) {
+    const listingPath = path.join(dirPath, listing);
+    const stat = fs.statSync(listingPath);
+    if (stat.isDirectory()) {
+      const dirDiagrams = findSvgDiagrams(listingPath);
+      diagrams.splice(diagrams.length, 0, ...dirDiagrams);
```
Comment (horsmand) [inline packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: This is fine by me to leave as-is, but I'm curious why you'd use `splice` over `concat`?

**Review by horsmand — CHANGES_REQUESTED** [review]

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts (hunk 1 above)
Comment (jusiskin) [reply packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: good catch. fixed

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts (hunk 2 above)
Comment (jusiskin) [reply packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: fixed

File: packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts (hunk 3 above)
Comment (jusiskin) [reply packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: no reason. concat is more readable, so I switched it over

Comment (horsmand) [reply packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts]: I think `concat` works without needing the `...` but I don't think it makes a difference.

**Review by horsmand — APPROVED** [review]

**Review by ddneilson — APPROVED** [review]
This is great, thanks Josh! The added test is a great idea!

**Pull request merged**